src/main/scala/com/dwolla/lambda/cloudformation/AbstractCustomResourceHandler.scala

```diff
@@ -1,82 +1,103 @@
 package com.dwolla.lambda.cloudformation
 
-import java.io.{InputStream, OutputStream, PrintWriter, StringWriter}
+import java.io._
 
-import com.amazonaws.services.lambda.runtime.{Context, RequestStreamHandler}
+import cats.data._
+import cats.effect._
+import cats.implicits._
+import com.amazonaws.services.lambda.runtime._
 import com.dwolla.lambda.cloudformation.AbstractCustomResourceHandler.stackTraceLines
-import org.json4s.ParserUtil.ParseException
-import org.json4s._
-import org.json4s.native.Serialization._
-import org.slf4j.{Logger, LoggerFactory}
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.parser._
+import org.slf4j._
 
-import scala.concurrent.duration._
-import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent._
 import scala.io.Source
-import scala.language.postfixOps
-import scala.util.Try
+import scala.language.higherKinds
 
-abstract class AbstractCustomResourceHandler extends RequestStreamHandler {
+abstract class AbstractCustomResourceHandler extends CatsAbstractCustomResourceHandler[IO] {
```

Comment:
Should this be marked `deprecated` as well?

Reply:
It's possible I don't understand how this will be used, but if I were to write a handler that extended `AbstractCustomResourceHandler`, I would have to implement `createParsedRequestHandler()` which returns `ParsedCloudFormationCustomResourceRequestHandler`.  The return value is deprecated and I'm not sure what a implementer would do to resolve the deprecation warning.

Reply:
Good point. I'll deprecate `AbstractCustomResourceHandler` too.

Reply:
Addressed in eafa3a5